File: Backend/ClassicTextileServer/src/main/java/com/example/ClassicTextile/modules/Order.java
package com.example.ClassicTextile.modules;

import com.example.ClassicTextile.modules.enums.OrderStatus;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Fetch;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "Orders")
@Data
@NoArgsConstructor
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(value = AccessLevel.NONE)
    private int id;
    @ManyToOne
    @JsonIgnore
    private User user;
    private LocalDate orderDate;
    private double totalPrice;
    private OrderStatus orderStatus=OrderStatus.PENDING;
    private String shippingAddress;
    @OneToMany(mappedBy = "order",fetch = FetchType.EAGER , orphanRemoval = true)
    private List<OrderItem> orderItems;

    public Order(User user, LocalDate orderDate, double totalPrice, String shippingAddress, List<OrderItem> orderItems) {
        this.user = user;
        this.orderDate = orderDate;
        this.totalPrice = totalPrice;
        this.shippingAddress = shippingAddress;
        this.orderItems = orderItems;
    }

    public Order(User user, LocalDate orderDate, List<OrderItem> orderItems) {
        this.user = user;
        this.orderDate = orderDate;
        this.orderItems = orderItems;
    }

    @Override
    public String toString() {
        return "Order{" +
                "user=" + user.getId() +
                ", orderDate=" + orderDate +
                ", totalPrice=" + totalPrice +
                ", orderStatus=" + orderStatus +
                ", shippingAddress='" + shippingAddress + '\'' +
                ", orderItems=" + orderItems +
                '}';
    }
}

File: Frontend/classic_app/src/Models/Order.ts
import { OrderStatus } from "./enums/OrderStatus"
import { OrderItem } from "./OrderItem"
import { User } from "./User"

export class Order{
    id:number
    user:User
    orderDate:Date
    totalPrice:number
    orderStatus:OrderStatus
    shippingAddress:string
    orderItems:OrderItem[]

    constructor(id:number,user:User,orderDate:Date,totalPrice:number,orderStatus:OrderStatus,shippingAddress:string,orderItems:OrderItem[]){
            this.id=id
            this.user=user
            this.totalPrice=totalPrice
            this.orderDate=orderDate
            this.orderStatus=orderStatus
            this.shippingAddress=shippingAddress
            this.orderItems=orderItems
        }
}